File: Redstone/Types/Difficulty.cs
namespace Redstone.Types
{
    /// <summary>
    /// Minecraft's difficulty levels.
    /// </summary>
    public enum Difficulty : byte
    {
        /// <summary>
        /// Safest and easiest mode. There are no hostile mobs, with the
        /// exception of the Ender Dragon. Damage only comes from the
        /// physical world. Food is not required.
        /// </summary>
        Peaceful = 0,

        /// <summary>
        /// Easy mode has mobs. Food is depleted, however you don't die from
        /// starvation.
        /// </summary>
        Easy = 1,

        /// <summary>
        /// The standard mode of Minecraft. Damage is harder. Still cannot
        /// die from starvation.
        /// </summary>
        Normal = 2,

        /// <summary>
        /// The hardest mode of Minecraft. You may die from starvation and
        /// mobs are the hardest.
        /// </summary>
        Hard = 3
    }
}

File: Minecraft Types/src/com/CopperPenguin96/MinecraftTypes/Difficulty.java
package com.CopperPenguin96.MinecraftTypes;

/***
 * 
 * @author CopperPenguin96
 *
 */
public enum Difficulty {
	
	Peaceful(0),
	Easy(1),
	Normal(2),
	Hard(3);

	private int _difficulty = 2;
	
	public int getValue() {
		return _difficulty;
	}
	
	public void setValue(int i) {
		if (i > 3) {
			throw new IndexOutOfBoundsException();
		}
		
		_difficulty = i;
	}
	
	Difficulty(int i) {
		setValue(i);
	}
	
}